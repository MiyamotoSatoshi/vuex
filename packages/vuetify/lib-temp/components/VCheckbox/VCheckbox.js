// Styles
import '../../stylus/components/_selection-controls.styl';
// Components
import VIcon from '../VIcon';
// import { VFadeTransition } from '../transitions'
// Mixins
import Selectable from '../../mixins/selectable';
/* @vue/component */
export default {
    name: 'v-checkbox',
    mixins: [
        Selectable
    ],
    props: {
        indeterminate: Boolean,
        indeterminateIcon: {
            type: String,
            default: '$vuetify.icons.checkboxIndeterminate'
        },
        onIcon: {
            type: String,
            default: '$vuetify.icons.checkboxOn'
        },
        offIcon: {
            type: String,
            default: '$vuetify.icons.checkboxOff'
        }
    },
    data: vm => ({
        inputIndeterminate: vm.indeterminate
    }),
    computed: {
        classes() {
            return {
                'v-input--selection-controls': true,
                'v-input--checkbox': true
            };
        },
        computedIcon() {
            if (this.inputIndeterminate) {
                return this.indeterminateIcon;
            }
            else if (this.isActive) {
                return this.onIcon;
            }
            else {
                return this.offIcon;
            }
        }
    },
    watch: {
        indeterminate(val) {
            this.inputIndeterminate = val;
        }
    },
    methods: {
        genCheckbox() {
            return this.$createElement('div', {
                staticClass: 'v-input--selection-controls__input'
            }, [
                this.genInput('checkbox', {
                    ...this.$attrs,
                    'aria-checked': this.inputIndeterminate
                        ? 'mixed'
                        : this.isActive.toString()
                }),
                !this.disabled && this.genRipple(this.setTextColor(this.computedColor)),
                this.$createElement(VIcon, this.setTextColor(this.computedColor, {
                    props: {
                        dark: this.dark,
                        light: this.light
                    }
                }), this.computedIcon)
            ]);
        },
        genDefaultSlot() {
            return [
                this.genCheckbox(),
                this.genLabel()
            ];
        }
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVkNoZWNrYm94LmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiLi4vLi4vLi4vc3JjL2NvbXBvbmVudHMvVkNoZWNrYm94L1ZDaGVja2JveC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxTQUFTO0FBQ1QsT0FBTyxrREFBa0QsQ0FBQTtBQUV6RCxhQUFhO0FBQ2IsT0FBTyxLQUFLLE1BQU0sVUFBVSxDQUFBO0FBQzVCLG1EQUFtRDtBQUVuRCxTQUFTO0FBQ1QsT0FBTyxVQUFVLE1BQU0seUJBQXlCLENBQUE7QUFFaEQsb0JBQW9CO0FBQ3BCLGVBQWU7SUFDYixJQUFJLEVBQUUsWUFBWTtJQUVsQixNQUFNLEVBQUU7UUFDTixVQUFVO0tBQ1g7SUFFRCxLQUFLLEVBQUU7UUFDTCxhQUFhLEVBQUUsT0FBTztRQUN0QixpQkFBaUIsRUFBRTtZQUNqQixJQUFJLEVBQUUsTUFBTTtZQUNaLE9BQU8sRUFBRSxzQ0FBc0M7U0FDaEQ7UUFDRCxNQUFNLEVBQUU7WUFDTixJQUFJLEVBQUUsTUFBTTtZQUNaLE9BQU8sRUFBRSwyQkFBMkI7U0FDckM7UUFDRCxPQUFPLEVBQUU7WUFDUCxJQUFJLEVBQUUsTUFBTTtZQUNaLE9BQU8sRUFBRSw0QkFBNEI7U0FDdEM7S0FDRjtJQUVELElBQUksRUFBRSxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUM7UUFDWCxrQkFBa0IsRUFBRSxFQUFFLENBQUMsYUFBYTtLQUNyQyxDQUFDO0lBRUYsUUFBUSxFQUFFO1FBQ1IsT0FBTztZQUNMLE9BQU87Z0JBQ0wsNkJBQTZCLEVBQUUsSUFBSTtnQkFDbkMsbUJBQW1CLEVBQUUsSUFBSTthQUMxQixDQUFBO1FBQ0gsQ0FBQztRQUNELFlBQVk7WUFDVixJQUFJLElBQUksQ0FBQyxrQkFBa0IsRUFBRTtnQkFDM0IsT0FBTyxJQUFJLENBQUMsaUJBQWlCLENBQUE7YUFDOUI7aUJBQU0sSUFBSSxJQUFJLENBQUMsUUFBUSxFQUFFO2dCQUN4QixPQUFPLElBQUksQ0FBQyxNQUFNLENBQUE7YUFDbkI7aUJBQU07Z0JBQ0wsT0FBTyxJQUFJLENBQUMsT0FBTyxDQUFBO2FBQ3BCO1FBQ0gsQ0FBQztLQUNGO0lBRUQsS0FBSyxFQUFFO1FBQ0wsYUFBYSxDQUFFLEdBQUc7WUFDaEIsSUFBSSxDQUFDLGtCQUFrQixHQUFHLEdBQUcsQ0FBQTtRQUMvQixDQUFDO0tBQ0Y7SUFFRCxPQUFPLEVBQUU7UUFDUCxXQUFXO1lBQ1QsT0FBTyxJQUFJLENBQUMsY0FBYyxDQUFDLEtBQUssRUFBRTtnQkFDaEMsV0FBVyxFQUFFLG9DQUFvQzthQUNsRCxFQUFFO2dCQUNELElBQUksQ0FBQyxRQUFRLENBQUMsVUFBVSxFQUFFO29CQUN4QixHQUFHLElBQUksQ0FBQyxNQUFNO29CQUNkLGNBQWMsRUFBRSxJQUFJLENBQUMsa0JBQWtCO3dCQUNyQyxDQUFDLENBQUMsT0FBTzt3QkFDVCxDQUFDLENBQUMsSUFBSSxDQUFDLFFBQVEsQ0FBQyxRQUFRLEVBQUU7aUJBQzdCLENBQUM7Z0JBQ0YsQ0FBQyxJQUFJLENBQUMsUUFBUSxJQUFJLElBQUksQ0FBQyxTQUFTLENBQUMsSUFBSSxDQUFDLFlBQVksQ0FBQyxJQUFJLENBQUMsYUFBYSxDQUFDLENBQUM7Z0JBQ3ZFLElBQUksQ0FBQyxjQUFjLENBQUMsS0FBSyxFQUFFLElBQUksQ0FBQyxZQUFZLENBQUMsSUFBSSxDQUFDLGFBQWEsRUFBRTtvQkFDL0QsS0FBSyxFQUFFO3dCQUNMLElBQUksRUFBRSxJQUFJLENBQUMsSUFBSTt3QkFDZixLQUFLLEVBQUUsSUFBSSxDQUFDLEtBQUs7cUJBQ2xCO2lCQUNGLENBQUMsRUFBRSxJQUFJLENBQUMsWUFBWSxDQUFDO2FBQ3ZCLENBQUMsQ0FBQTtRQUNKLENBQUM7UUFDRCxjQUFjO1lBQ1osT0FBTztnQkFDTCxJQUFJLENBQUMsV0FBVyxFQUFFO2dCQUNsQixJQUFJLENBQUMsUUFBUSxFQUFFO2FBQ2hCLENBQUE7UUFDSCxDQUFDO0tBQ0Y7Q0FDRixDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiLy8gU3R5bGVzXG5pbXBvcnQgJy4uLy4uL3N0eWx1cy9jb21wb25lbnRzL19zZWxlY3Rpb24tY29udHJvbHMuc3R5bCdcblxuLy8gQ29tcG9uZW50c1xuaW1wb3J0IFZJY29uIGZyb20gJy4uL1ZJY29uJ1xuLy8gaW1wb3J0IHsgVkZhZGVUcmFuc2l0aW9uIH0gZnJvbSAnLi4vdHJhbnNpdGlvbnMnXG5cbi8vIE1peGluc1xuaW1wb3J0IFNlbGVjdGFibGUgZnJvbSAnLi4vLi4vbWl4aW5zL3NlbGVjdGFibGUnXG5cbi8qIEB2dWUvY29tcG9uZW50ICovXG5leHBvcnQgZGVmYXVsdCB7XG4gIG5hbWU6ICd2LWNoZWNrYm94JyxcblxuICBtaXhpbnM6IFtcbiAgICBTZWxlY3RhYmxlXG4gIF0sXG5cbiAgcHJvcHM6IHtcbiAgICBpbmRldGVybWluYXRlOiBCb29sZWFuLFxuICAgIGluZGV0ZXJtaW5hdGVJY29uOiB7XG4gICAgICB0eXBlOiBTdHJpbmcsXG4gICAgICBkZWZhdWx0OiAnJHZ1ZXRpZnkuaWNvbnMuY2hlY2tib3hJbmRldGVybWluYXRlJ1xuICAgIH0sXG4gICAgb25JY29uOiB7XG4gICAgICB0eXBlOiBTdHJpbmcsXG4gICAgICBkZWZhdWx0OiAnJHZ1ZXRpZnkuaWNvbnMuY2hlY2tib3hPbidcbiAgICB9LFxuICAgIG9mZkljb246IHtcbiAgICAgIHR5cGU6IFN0cmluZyxcbiAgICAgIGRlZmF1bHQ6ICckdnVldGlmeS5pY29ucy5jaGVja2JveE9mZidcbiAgICB9XG4gIH0sXG5cbiAgZGF0YTogdm0gPT4gKHtcbiAgICBpbnB1dEluZGV0ZXJtaW5hdGU6IHZtLmluZGV0ZXJtaW5hdGVcbiAgfSksXG5cbiAgY29tcHV0ZWQ6IHtcbiAgICBjbGFzc2VzICgpIHtcbiAgICAgIHJldHVybiB7XG4gICAgICAgICd2LWlucHV0LS1zZWxlY3Rpb24tY29udHJvbHMnOiB0cnVlLFxuICAgICAgICAndi1pbnB1dC0tY2hlY2tib3gnOiB0cnVlXG4gICAgICB9XG4gICAgfSxcbiAgICBjb21wdXRlZEljb24gKCkge1xuICAgICAgaWYgKHRoaXMuaW5wdXRJbmRldGVybWluYXRlKSB7XG4gICAgICAgIHJldHVybiB0aGlzLmluZGV0ZXJtaW5hdGVJY29uXG4gICAgICB9IGVsc2UgaWYgKHRoaXMuaXNBY3RpdmUpIHtcbiAgICAgICAgcmV0dXJuIHRoaXMub25JY29uXG4gICAgICB9IGVsc2Uge1xuICAgICAgICByZXR1cm4gdGhpcy5vZmZJY29uXG4gICAgICB9XG4gICAgfVxuICB9LFxuXG4gIHdhdGNoOiB7XG4gICAgaW5kZXRlcm1pbmF0ZSAodmFsKSB7XG4gICAgICB0aGlzLmlucHV0SW5kZXRlcm1pbmF0ZSA9IHZhbFxuICAgIH1cbiAgfSxcblxuICBtZXRob2RzOiB7XG4gICAgZ2VuQ2hlY2tib3ggKCkge1xuICAgICAgcmV0dXJuIHRoaXMuJGNyZWF0ZUVsZW1lbnQoJ2RpdicsIHtcbiAgICAgICAgc3RhdGljQ2xhc3M6ICd2LWlucHV0LS1zZWxlY3Rpb24tY29udHJvbHNfX2lucHV0J1xuICAgICAgfSwgW1xuICAgICAgICB0aGlzLmdlbklucHV0KCdjaGVja2JveCcsIHtcbiAgICAgICAgICAuLi50aGlzLiRhdHRycyxcbiAgICAgICAgICAnYXJpYS1jaGVja2VkJzogdGhpcy5pbnB1dEluZGV0ZXJtaW5hdGVcbiAgICAgICAgICAgID8gJ21peGVkJ1xuICAgICAgICAgICAgOiB0aGlzLmlzQWN0aXZlLnRvU3RyaW5nKClcbiAgICAgICAgfSksXG4gICAgICAgICF0aGlzLmRpc2FibGVkICYmIHRoaXMuZ2VuUmlwcGxlKHRoaXMuc2V0VGV4dENvbG9yKHRoaXMuY29tcHV0ZWRDb2xvcikpLFxuICAgICAgICB0aGlzLiRjcmVhdGVFbGVtZW50KFZJY29uLCB0aGlzLnNldFRleHRDb2xvcih0aGlzLmNvbXB1dGVkQ29sb3IsIHtcbiAgICAgICAgICBwcm9wczoge1xuICAgICAgICAgICAgZGFyazogdGhpcy5kYXJrLFxuICAgICAgICAgICAgbGlnaHQ6IHRoaXMubGlnaHRcbiAgICAgICAgICB9XG4gICAgICAgIH0pLCB0aGlzLmNvbXB1dGVkSWNvbilcbiAgICAgIF0pXG4gICAgfSxcbiAgICBnZW5EZWZhdWx0U2xvdCAoKSB7XG4gICAgICByZXR1cm4gW1xuICAgICAgICB0aGlzLmdlbkNoZWNrYm94KCksXG4gICAgICAgIHRoaXMuZ2VuTGFiZWwoKVxuICAgICAgXVxuICAgIH1cbiAgfVxufVxuIl19