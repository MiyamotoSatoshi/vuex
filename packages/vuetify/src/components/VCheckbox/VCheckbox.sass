@use 'sass:map'
@use '../../styles/settings'
@use '../../styles/tools'
@use './variables' as *

.v-checkbox
  .v-selection-control
    height: var(--v-input-control-height)
