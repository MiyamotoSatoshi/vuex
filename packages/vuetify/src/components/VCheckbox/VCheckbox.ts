// Styles
import '../../stylus/components/_selection-controls.styl'

// Components
import VIcon from '../VIcon'

// Mixins
import Selectable from '../../mixins/selectable'

/* @vue/component */
export default Selectable.extend({
  name: 'v-checkbox',

  props: {
    indeterminate: Boolean,
    indeterminateIcon: {
      type: String,
      default: '$vuetify.icons.checkboxIndeterminate'
    },
    onIcon: {
      type: String,
      default: '$vuetify.icons.checkboxOn'
    },
    offIcon: {
      type: String,
      default: '$vuetify.icons.checkboxOff'
    }
  },

  data () {
    return {
      inputIndeterminate: this.indeterminate
    }
  },

  computed: {
    classes (): object {
      return {
        'v-input--selection-controls': true,
        'v-input--checkbox': true
      }
    },
    computedIcon (): string {
      if (this.inputIndeterminate) {
        return this.indeterminateIcon
      } else if (this.isActive) {
        return this.onIcon
      } else {
        return this.offIcon
      }
    }
  },

  watch: {
    indeterminate (val) {
      this.inputIndeterminate = val
    }
  },

  methods: {
    genCheckbox () {
      return this.$createElement('div', {
        staticClass: 'v-input--selection-controls__input'
      }, [
        this.genInput('checkbox', {
          ...this.$attrs,
          'aria-checked': this.inputIndeterminate
            ? 'mixed'
            : this.isActive.toString()
        }),
        this.genRipple(this.setTextColor(this.computedColor)),
        this.$createElement(VIcon, this.setTextColor(this.computedColor, {
          props: {
            dark: this.dark,
            light: this.light
          }
        }), this.computedIcon)
      ])
    },
    genDefaultSlot () {
      return [
        this.genCheckbox(),
        this.genLabel()
      ]
    }
  }
})
