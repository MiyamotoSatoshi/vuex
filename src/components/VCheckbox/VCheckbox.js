import '../../stylus/components/_input-groups.styl'
import '../../stylus/components/_selection-controls.styl'

import VIcon from '../VIcon'
import { VFadeTransition } from '../transitions'
import Rippleable from '../../mixins/rippleable'
import Selectable from '../../mixins/selectable'

export default {
  name: 'v-checkbox',

  components: {
    VFadeTransition,
    VIcon
  },

  mixins: [Rippleable, Selectable],

  data () {
    return {
      inputIndeterminate: this.indeterminate
    }
  },

  props: {
    indeterminate: Boolean
  },

  computed: {
    classes () {
      const classes = {
        'checkbox': true,
        'input-group--selection-controls': true,
        'input-group--active': this.isActive
      }

      if (this.hasError) {
        classes['error--text'] = true
      } else {
        return this.addTextColorClassChecks(classes)
      }

      return classes
    },
    icon () {
      if (this.inputIndeterminate) {
        return 'indeterminate_check_box'
      } else if (this.isActive) {
        return 'check_box'
      } else {
        return 'check_box_outline_blank'
      }
    }
  },

  methods: {
    groupFocus (e) {
      this.isFocused = true
      this.$emit('focus', e)
    },
    groupBlur (e) {
      this.isFocused = false
      this.tabFocused = false
      this.$emit('blur', this.inputValue)
    }
  },

  render (h) {
    const transition = h('v-fade-transition', [
      h('v-icon', {
        staticClass: 'icon--selection-control',
        'class': {
          'icon--checkbox': this.icon === 'check_box'
        },
        key: this.icon,
        on: Object.assign({
          click: this.toggle
        }, this.$listeners)
      }, this.icon)
    ])

    const data = {
      attrs: {
        tabindex: this.disabled
          ? -1
          : this.internalTabIndex || this.tabindex,
        role: 'checkbox',
        'aria-checked': this.inputIndeterminate ? 'mixed' : (this.isActive ? 'true' : 'false'),
        'aria-label': this.label
      }
    }

    const ripple = this.ripple ? this.genRipple() : null

    return this.genInputGroup([transition, ripple], data)
  }
}
