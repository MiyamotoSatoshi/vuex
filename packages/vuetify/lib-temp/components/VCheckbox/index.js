import VCheckbox from './VCheckbox';
export { VCheckbox };
export default VCheckbox;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WQ2hlY2tib3gvaW5kZXguanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsT0FBTyxTQUFTLE1BQU0sYUFBYSxDQUFBO0FBRW5DLE9BQU8sRUFBRSxTQUFTLEVBQUUsQ0FBQTtBQUNwQixlQUFlLFNBQVMsQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBWQ2hlY2tib3ggZnJvbSAnLi9WQ2hlY2tib3gnXG5cbmV4cG9ydCB7IFZDaGVja2JveCB9XG5leHBvcnQgZGVmYXVsdCBWQ2hlY2tib3hcbiJdfQ==