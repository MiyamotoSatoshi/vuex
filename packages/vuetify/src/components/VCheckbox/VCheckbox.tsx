// Styles
import './VCheckbox.sass'

// Components
import { filterInputProps, makeVInputProps, VInput } from '@/components/VInput/VInput'
import { filterControlProps, makeSelectionControlProps, VSelectionControl } from '@/components/VSelectionControl/VSelectionControl'

// Composables
import { useProxiedModel } from '@/composables/proxiedModel'

// Utility
import { computed } from 'vue'
import { defineComponent, filterInputAttrs, useRender } from '@/util'

export const VCheckbox = defineComponent({
  name: 'VCheckbox',

  inheritAttrs: false,

  props: {
    indeterminate: Boolean,
    indeterminateIcon: {
      type: String,
      default: '$checkboxIndeterminate',
    },

    ...makeVInputProps(),
    ...makeSelectionControlProps(),

    falseIcon: {
      type: String,
      default: '$checkboxOff',
    },
    trueIcon: {
      type: String,
      default: '$checkboxOn',
    },
  },

  emits: {
    'update:indeterminate': (val: boolean) => true,
  },

  setup (props, { attrs, slots }) {
    const indeterminate = useProxiedModel(props, 'indeterminate')
    const falseIcon = computed(() => {
      return indeterminate.value
        ? props.indeterminateIcon
        : props.falseIcon
    })
    const trueIcon = computed(() => {
      return indeterminate.value
        ? props.indeterminateIcon
        : props.trueIcon
    })

    function onChange () {
      if (indeterminate.value) {
        indeterminate.value = false
      }
    }

    useRender(() => {
      const [inputAttrs, controlAttrs] = filterInputAttrs(attrs)
      const [inputProps, _1] = filterInputProps(props)
      const [controlProps, _2] = filterControlProps(props)

      return (
        <VInput
          class="v-checkbox"
          { ...inputAttrs }
          { ...inputProps }
        >
          {{
            ...slots,
            default: ({
              isDisabled,
              isReadonly,
            }) => (
              <VSelectionControl
                { ...controlProps }
                type="checkbox"
                onUpdate:modelValue={ onChange }
                falseIcon={ falseIcon.value }
                trueIcon={ trueIcon.value }
                aria-checked={ indeterminate.value ? 'mixed' : undefined }
                disabled={ isDisabled.value }
                readonly={ isReadonly.value }
                { ...controlAttrs }
                v-slots={ slots }
              />
            ),
          }}
        </VInput>
      )
    })

    return {}
  },
})

export type VCheckbox = InstanceType<typeof VCheckbox>
