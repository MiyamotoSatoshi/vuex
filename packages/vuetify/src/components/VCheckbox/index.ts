import VCheckbox from './VCheckbox'
import VSimpleCheckbox from './VSimpleCheckbox'

export { VCheckbox, VSimpleCheckbox }
export default VCheckbox
