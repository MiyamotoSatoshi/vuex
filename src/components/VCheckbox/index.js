import VCheckbox from './VCheckbox'

/* istanbul ignore next */
VCheckbox.install = function install (Vue) {
  Vue.component(VCheckbox.name, VCheckbox)
}

export { VCheckbox }
export default VCheckbox
