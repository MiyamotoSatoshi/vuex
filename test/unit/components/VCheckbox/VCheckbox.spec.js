import { test } from '@/test'
import VCheckbox from '@/components/VCheckbox'

test('VCheckbox.js', ({ mount }) => {
  it('should return true when clicked', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        inputValue: false
      }
    })

    const input = wrapper.first('input')

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    input.trigger('click')
    expect(change).toBeCalledWith(true)
  })

  it('should return a value when toggled on with a specified value', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        value: 'John',
        inputValue: null
      }
    })

    const input = wrapper.first('input')

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    input.trigger('click')
    expect(change).toBeCalledWith('John')
  })

  it('should return null when toggled off with a specified value', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        value: 'John',
        inputValue: 'John'
      }
    })

    const ripple = wrapper.find('input')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    ripple.trigger('click')
    expect(change).toBeCalledWith(null)
  })

  it('should toggle when label is clicked', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        label: 'Label',
        value: null
      },
      attrs: {}
    })

    const label = wrapper.find('label')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    label.trigger('click')
    expect(change).toBeCalled()
  })

  it('should render role and aria-checked attributes on input group', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        inputValue: false
      }
    })

    const input = wrapper.first('input')

    expect(input.getAttribute('role')).toBe('checkbox')
    expect(input.getAttribute('aria-checked')).toBe('false')

    wrapper.setProps({ 'inputValue': true })
    expect(input.getAttribute('aria-checked')).toBe('true')

    wrapper.setProps({ 'indeterminate': true })
    expect(input.getAttribute('aria-checked')).toBe('mixed')
  })

  it('should render aria-label attribute with label value on input group', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        label: 'Test'
      },
      attrs: {}
    })

    const inputGroup = wrapper.first('input')
    expect(inputGroup.getAttribute('aria-label')).toBe('Test')
  })

  it('should not render aria-label attribute with no label value on input group', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        label: null
      }
    })

    const inputGroup = wrapper.first('input')
    expect(inputGroup.element.getAttribute('aria-label')).toBeFalsy()
  })

  it('should toggle on space and enter with default toggleKeys', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        inputValue: false
      }
    })

    const change = jest.fn()
    wrapper.vm.$on('change', change)
    const input = wrapper.first('input')

    input.trigger('keydown.enter')
    input.trigger('keydown.space')

    expect(change.mock.calls).toHaveLength(2)
  })

  it('should not toggle on space or enter with blank toggleKeys', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        inputValue: false,
        toggleKeys: []
      }
    })

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    wrapper.trigger('keydown.enter')
    wrapper.trigger('keydown.space')

    expect(change).not.toBeCalled()
  })

  it('should toggle only on custom toggleKeys', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        inputValue: false,
        toggleKeys: [32] // space
      }
    })

    const change = jest.fn()
    wrapper.vm.$on('change', change)
    const input = wrapper.first('input')

    input.trigger('keydown.enter')
    expect(change).not.toBeCalled()

    input.trigger('keydown.space')
    expect(change).toBeCalled()
  })

  it('should enable ripple based on disabled state', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        inputValue: false,
        disabled: false
      }
    })

    const ripple = wrapper.first('.v-input--selection-controls__ripple')

    expect(ripple.element._ripple.enabled).toBe(true)
    expect(ripple.element._ripple.centered).toBe(true)

    wrapper.setProps({ disabled: true })

    expect(ripple.element._ripple.enabled).toBe(false)
  })

  it('should set ripple centered property when enabled', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        inputValue: false,
        disabled: true
      }
    })

    const ripple = wrapper.first('.v-input--selection-controls__ripple')

    expect(ripple.element._ripple.enabled).toBe(false)

    wrapper.setProps({ disabled: false })

    expect(ripple.element._ripple.enabled).toBe(true)
    expect(ripple.element._ripple.centered).toBe(true)
  })

  it('should not render ripple when ripple prop is false', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        inputValue: false,
        ripple: false
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')

    expect(ripple).toHaveLength(0)
  })

  it('should render ripple when ripple prop is true', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        ripple: true
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    expect(ripple.element._ripple.enabled).toBe(true)
    expect(ripple.element._ripple.centered).toBe(true)
  })

  it('should return a value when toggled on with a specified object value', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        value: {x: 1, y: 2},
        inputValue: null
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    ripple.trigger('click')
    expect(change).toBeCalledWith({x: 1, y: 2})
  })

  it('should return a value when toggled on with a specified array value', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        value: [1, "2", {x: 1, y: 2}],
        inputValue: null
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    ripple.trigger('click')
    expect(change).toBeCalledWith([1, "2", {x: 1, y: 2}])
  })

  it('should push value to array when toggled on and is multiple', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        value: 'John',
        inputValue: []
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    ripple.trigger('click')
    expect(change).toBeCalledWith(['John'])
  })

  it('should push array value to array when toggled on and is multiple', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        value: [1, 2, {x: 1, y: 2}],
        inputValue: ['Existing']
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    ripple.trigger('click')
    expect(change).toBeCalledWith(['Existing', [1, 2, {x: 1, y: 2}]])
  })

  it('should return null when toggled off with a specified array value', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        multiple: false, // must use multiple flag for array values
        value: ['John'],
        inputValue: ['John']
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    ripple.trigger('click')
    expect(change).toBeCalledWith(null)
  })

  it('should remove value(s) from array when toggled off and multiple', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        value: 1,
        inputValue: [1, 2, 1, 3]
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    ripple.trigger('click')
    expect(change).toBeCalledWith([2, 3])
  })

  it('should remove value(s) from array when toggled off and multiple - with objects', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        value: {a: 1},
        inputValue: [{a: 1}, {b: 1}, {a: 1}, {c: 1}]
      }
    })

    const ripple = wrapper.find('.v-input--selection-controls__ripple')[0]

    const change = jest.fn()
    wrapper.vm.$on('change', change)

    ripple.trigger('click')
    expect(change).toBeCalledWith([{b: 1}, {c: 1}])
  })

  // https://github.com/vuetifyjs/vuetify/issues/2119
  it('should put id on internal input', () => {
    const wrapper = mount(VCheckbox, {
      propsData: { id: 'foo' }
    })

    const input = wrapper.first('input')
    expect(input.element.id).toBe('foo')
  })

  it('should use custom icons', () => {
    const wrapper = mount(VCheckbox, {
      propsData: {
        indeterminateIcon: 'fizzbuzz',
        onIcon: 'foo',
        offIcon: 'bar',
        indeterminate: true,
        value: 'fizz'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
    wrapper.setData({ value: 'fizz'})
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.setData({ value: 'buzz'})
    expect(wrapper.html()).toMatchSnapshot()
  })
})
