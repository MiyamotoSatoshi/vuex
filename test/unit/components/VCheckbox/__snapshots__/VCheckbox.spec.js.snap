// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCheckbox.js should be disabled 1`] = `

<input aria-checked="false"
       disabled="disabled"
       role="checkbox"
       type="checkbox"
>

`;

exports[`VCheckbox.js should render themed component 1`] = `

<div class="v-input v-input--selection-controls v-input--checkbox theme--light">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-input--selection-controls__input">
        <input aria-checked="false"
               role="checkbox"
               type="checkbox"
        >
        <div class="v-input--selection-controls__ripple">
        </div>
        <i aria-hidden="true"
           class="v-icon theme--light material-icons"
        >
          check_box_outline_blank
        </i>
      </div>
    </div>
    <div class="v-messages theme--light">
      <div class="v-messages__wrapper">
      </div>
    </div>
  </div>
</div>

`;

exports[`VCheckbox.js should use custom icons 1`] = `

<div class="v-input v-input--selection-controls v-input--checkbox">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-input--selection-controls__input">
        <input aria-checked="mixed"
               role="checkbox"
               type="checkbox"
        >
        <div class="v-input--selection-controls__ripple">
        </div>
        <i aria-hidden="true"
           class="v-icon material-icons"
        >
          fizzbuzz
        </i>
      </div>
    </div>
    <div class="v-messages">
      <div class="v-messages__wrapper">
      </div>
    </div>
  </div>
</div>

`;

exports[`VCheckbox.js should use custom icons 2`] = `

<div class="v-input v-input--selection-controls v-input--checkbox">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-input--selection-controls__input">
        <input aria-checked="mixed"
               role="checkbox"
               type="checkbox"
        >
        <div class="v-input--selection-controls__ripple">
        </div>
        <i aria-hidden="true"
           class="v-icon material-icons"
        >
          fizzbuzz
        </i>
      </div>
    </div>
    <div class="v-messages">
      <div class="v-messages__wrapper">
      </div>
    </div>
  </div>
</div>

`;

exports[`VCheckbox.js should use custom icons 3`] = `

<div class="v-input v-input--selection-controls v-input--checkbox">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-input--selection-controls__input">
        <input aria-checked="mixed"
               role="checkbox"
               type="checkbox"
        >
        <div class="v-input--selection-controls__ripple">
        </div>
        <i aria-hidden="true"
           class="v-icon material-icons"
        >
          fizzbuzz
        </i>
      </div>
    </div>
    <div class="v-messages">
      <div class="v-messages__wrapper">
      </div>
    </div>
  </div>
</div>

`;
